.github/workflows/test-levels-composer.patch

```diff
@@ -0,0 +1,18 @@
+diff --git a/composer.json b/composer.json
+index 6dde20b5f..7ea244ee1 100644
+--- a/composer.json
++++ b/composer.json
+@@ -116,8 +116,11 @@
+ 			]
+ 		},
+ 		"classmap": [
+-			"tests/e2e",
+-			"tests/PHPStan"
++			"tests/PHPStan/Rules/Methods/data/returnTypes.php",
```

Comment:
I understand the intention, but I don't love it. It's going to be really hard to maintain this list and debug problems as time goes on.

I'd be more interested in parallelizing running the tests. Basically each "topic" in LevelsTestCase could have its own GitHub Actions job.

It's not exactly the same but the idea is here: https://rias.be/blog/running-phpunit-tests-in-parallel-using-github-actions

Reply:
I have adjusted the PR. It now filters the existing testes based on class-name and builds a dynamic matrix with it.

this means we run now 40 tests in parallel (which each takes ~50s) instead of one large job which runs 10 minutes

Reply:
> I understand the intention, but I don't love it. It's going to be really hard to maintain this list and debug problems as time goes on.
> 
> I'd be more interested in parallelizing running the tests. Basically each "topic" in LevelsTestCase could have its own GitHub Actions job.
> 
> It's not exactly the same but the idea is here: [rias.be/blog/running-phpunit-tests-in-parallel-using-github-actions](https://rias.be/blog/running-phpunit-tests-in-parallel-using-github-actions)

This is very similar to what i did in psalm years ago:
https://github.com/vimeo/psalm/pull/4985/files

But in psalm, each job also runs test in parallel.

Reply:
@bendavies the tests jobs - even before this PR - are already highly parallelized - within the github action jobs and also across several jobs. 

This PR splits up one very slow part of the pre-existing tests into even more jobs because these were particular slow

Reply:
> i couldn't see (still can't) where tests are split across jobs.

see the screen in my above comment. there you can see the "levels test" matrix, with several jobs.
if you compare to other existing PR, this step is just a single job.

the PR implements this by inspecting the test-suite level-tests test-cases. Because of the used dataprovider, this means the `testLevels` method represents several test-cases. each of these test-cases is run in a separate github action job, because I feed this list of test-cases as a json array into the "test-levels" github action job matrix.